### DISCUSSION TITLE
GPG Key path

### DISCUSSION THREAD

**Discussion by bhaijiyunus** [ostreedev/ostree#3272] [Q&A] [answered]
Hi

Apart from gpg-import option, is there any other way to import the gpg key path to ostree remote? If yes then, how to do it?

Regards,
jbhaijy

**Comment by bhaijiyunus** [comment]
Can anyone help me to answer my question above? I am looking for help on adding the GPG key other than `--gpg-import` command line option.

I don't want to import the GPG key with `--gpg-import` option through command line on Linux based device. I am thinking to add the GPG key on its ostree trusted path i.e. /usr/share/ostree/trusted.gpg.d/ during the Yocto build time so that by default device will have GPG key where the ostree expects while verification.

**Reply by dbnicholson** [reply, marked as answer]
Sorry, I meant to reply to this initially. You can definitely put the keyring directly on disk. The import options are just helpers to serialize keys to the appropriate locations. At Endless we've been putting our main keyring in trusted.gpg.d for years. 

Unlike when importing, I'm pretty sure the on disk keyring needs to be unarmored. I.e., a raw stream of PGP packets. If you want to scope the keys to a particular remote, the path convention is `$repo_path/$remote_name.trusted.gpg`.
